packages/iris-grid/src/mousehandlers/IrisGridContextMenuHandler.tsx

```diff
@@ -531,70 +531,65 @@ class IrisGridContextMenuHandler extends GridMouseHandler {
     const { filterIconColor } = theme;
     const { settings } = irisGrid.props;
 
-    const allSelectedRanges = [...getSelectedRanges()];
+    let selectedRanges = [...getSelectedRanges()];
     // no selected range (i.e. right clicked a cell without highlighting it)
-    if (allSelectedRanges.length === 0) {
-      allSelectedRanges.push(
+    // although GridSelectionMouseHandler does change selectedRanges, state isn't updated in
+    //   time for getSelectedRanges to show the selected cell
+    if (selectedRanges.length === 0) {
+      selectedRanges.push(
         new GridRange(sourceColumn, sourceRow, sourceColumn, sourceRow)
       );
     }
 
-    // this block handles:
-    //  - truncates the ranges to search to MAX_MULTISELECT_VALUES
-    //    - then gets all the unique values
-    //    - this is instead of stopping at MAX unique values to prevent the situation when
-    //      there's a large amount of rows with little unique values
-    //  - checks if the source cell is in the selected ranges
-    //    - if it isn't, then the selected cell is the source cell instead
-    //    - although GridSelectionMouseHandler does change selectedRanges, that function and
-    //      this one run in the same cycle so state is not updated
-    let slicedSelectedRanges = [];
-    let sourceInSelected = false;
-    let valuesLeft = MAX_MULTISELECT_VALUES;
-    for (let i = 0; i < allSelectedRanges.length; i += 1) {
-      const selectedRange = allSelectedRanges[i];
-      if (selectedRange.startRow !== null && selectedRange.endRow !== null) {
-        // source cell in selected ranges
-        if (
-          selectedRange.startRow <= sourceRow &&
-          sourceRow <= selectedRange.endRow
-        ) {
-          sourceInSelected = true;
+    // - this block truncates the selected ranges to MAX_MULTISELECT_ROWS rows
+    //   - NOT first MAX_MULTISELECT_ROWS rows after the first row
+    //   - NOT first MAX_MULTISELECT_ROWS unique values (prevent case where there are a small
+    //     amount of values, but a large amount of rows with those values)
+    if (GridRange.containsCell(selectedRanges, sourceColumn, sourceRow)) {
+      let rowCount = GridRange.rowCount(selectedRanges);
+      while (rowCount > MAX_MULTISELECT_ROWS) {
+        const lastRow = selectedRanges.pop();
+        // should never occur, sanity check
+        if (lastRow === undefined) {
+          throw Error('Selected ranges should not be empty');
         }
```

Comment:
You can write this using an `assertNotNull` instead to make it a one-liner (need to add the import as well).
```suggestion
        assertNotNull(lastRow, 'Selected ranges should not be empty');
```